app/controllers/api/v1/plans_controller.rb

```diff
@@ -12,30 +12,11 @@ class PlansController < BaseApiController
       # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity
       # rubocop:disable Metrics/PerceivedComplexity, Metrics/MethodLength
       def show
-        plans = Plan.where(id: params[:id]).limit(1)
-
-        if plans.any?
-          if client.is_a?(User)
-            # If the specified plan does not belong to the org or the owner's org
-            if plans.first.org_id != client.org_id &&
-               plans.first.owner&.org_id != client.org_id
-
-              # Kaminari pagination requires an ActiveRecord resultset :/
-              plans = Plan.where(id: nil).limit(1)
-            end
-
-          elsif client.is_a?(ApiClient) && plans.first.api_client_id != client.id &&
-                !plans.first.publicly_visible?
-            # Kaminari pagination requires an ActiveRecord resultset :/
-            plans = Plan.where(id: nil).limit(1)
-          end
-
-          if plans.present? && plans.any?
-            @items = paginate_response(results: plans)
-            render "/api/v1/plans/index", status: :ok
-          else
-            render_error(errors: [_("Plan not found")], status: :not_found)
-          end
+        plans = Api::V1::PlansPolicy::Scope.new(client, Plan).resolve
+                                           .where(id: params[:id]).limit(1)
```

Comment:
It would be nice to re-use this pattern of using the policy-scopes to determine the viewership-rights across the app